### ISSUE TITLE
Compatibility with mtl-2.3

### ISSUE THREAD

**Issue by sjakobi** [kowainik/tomland#406]
The latest `tomland` release already seems to build with `mtl-2.3` and `--allow-newer`. Could you relax the constraint, e.g. with a Hackage revision?

**Comment by vrom911** [comment]
Resolved in #422

**Comment by andreabedini** [comment]
@vrom911 may I ask for a revision for tomland 1.3.3.2 and/or a new release?

The following dependency versions seem to work ok.

```
active-repositories: hackage.haskell.org:merge
constraints: any.array ==0.5.4.0,
             any.base ==4.17.1.0,
             any.binary ==0.8.9.1,
             any.bytestring ==0.11.4.0,
             any.case-insensitive ==1.2.1.0,
             any.containers ==0.6.7,
             any.deepseq ==1.4.8.0,
             any.filepath ==1.4.2.2,
             any.ghc-bignum ==1.3,
             any.ghc-boot-th ==9.4.5,
             any.ghc-prim ==0.9.0,
             any.hashable ==1.4.2.0,
             hashable +integer-gmp -random-initial-seed,
             any.integer-logarithms ==1.0.3.1,
             integer-logarithms -check-bounds +integer-gmp,
             any.megaparsec ==9.2.2,
             megaparsec -dev,
             any.mtl ==2.3.1,
             any.parser-combinators ==1.3.0,
             parser-combinators -dev,
             any.pretty ==1.1.3.6,
             any.primitive ==0.8.0.0,
             any.rts ==1.0.2,
             any.scientific ==0.3.7.0,
             scientific -bytestring-builder -integer-simple,
             any.selective ==0.5,
             any.template-haskell ==2.19.0.0,
             any.text ==2.0.2,
             any.time ==1.12.2,
             any.tomland ==1.3.3.2,
             tomland -build-play-tomland -build-readme,
             any.transformers ==0.5.6.2,
             any.unordered-containers ==0.2.19.1,
             unordered-containers -debug,
             any.validation-selective ==0.1.0.2
index-state: hackage.haskell.org 2023-05-25T01:39:18Z
```
